### ISSUE TITLE
Midpoint normalizer doesn't work well for data with large difference

### ISSUE THREAD

**Issue by zxdawn** [lukelbd/proplot#145]
### Description

Midpoint normalizer doesn't work well for data with large difference.
Zero value doesn't divide the colormap.

### Steps to reproduce

```python
import proplot as plot
import numpy as np

state = np.random.RandomState(51423)
data1 = (state.rand(20, 20) - 0.43).cumsum(axis=0)
data2 = (state.rand(20, 20) - 0.57).cumsum(axis=0)

data1[data1>2] *= 5
data2[data2<-3] *= 5

f, axs = plot.subplots(ncols=2, axwidth=2.5, aspect=1.5)
cmap = plot.Colormap('DryWet', cut=0.1)
axs.format(suptitle='Midpoint normalizer demo')
for ax, data, mode in zip(axs, (data1, data2), ('positive', 'negative')):
    m = ax.contourf(data, norm='midpoint', cmap='Div')
    ax.colorbar(m, loc='b')
    ax.format(title=f'Skewed {mode} data')
```

**Expected behavior**: [What you expected to happen]
![image](https://user-images.githubusercontent.com/30388627/80302161-ef8f7d00-87da-11ea-8044-e57ed06fc68c.png)

**Actual behavior**: [What actually happened]
![image](https://user-images.githubusercontent.com/30388627/80302045-4d6f9500-87da-11ea-9388-bab8fbf1f480.png)

### Equivalent steps in matplotlib

```python
import matplotlib
import matplotlib.pyplot as plt
import numpy as np

class MidpointNormalize(matplotlib.colors.Normalize):
    def __init__(self, vmin=None, vmax=None, midpoint=None, clip=False):
        self.midpoint = midpoint
        matplotlib.colors.Normalize.__init__(self, vmin, vmax, clip)

    def __call__(self, value, clip=None):
        # Note that I'm ignoring clipping and other edge cases here.
        result, is_scalar = self.process_value(value)
        x, y = [self.vmin, self.midpoint, self.vmax], [0, 0.5, 1]
        return np.ma.array(np.interp(value, x, y), mask=result.mask, copy=False)

state = np.random.RandomState(51423)
data1 = (state.rand(20, 20) - 0.43).cumsum(axis=0)
data2 = (state.rand(20, 20) - 0.57).cumsum(axis=0)

data1[data1>2] *= 5
data2[data2<-3] *= 5

f, axs = plt.subplots(1, 2)
for ax, data, mode in zip(axs, (data1, data2), ('positive', 'negative')):
    m = ax.contourf(data, norm=MidpointNormalize(midpoint=0), cmap='RdBu_r')
    plt.colorbar(m, ax=ax)
    ax.set_title(f'Skewed {mode} data')
```

### Proplot version

0.5.0

**Comment by lukelbd** [comment]
Thanks, it turned out this was due to an interaction between `MidpointNorm` and the [`BinNorm`](https://proplot.readthedocs.io/en/latest/api/proplot.styletools.BinNorm.html?) algorithm I use to generate discrete levels. To make sure the level colors use the full colormap range, it was scaling "negative" colors into the positive territory. This will be fixed in the next version.

**Comment by lukelbd** [comment]
This is fixed by 98a976f1, specifically these lines:

https://github.com/lukelbd/proplot/blob/98a976f1809f33aff055536e7d89ab3b19d85dd4/proplot/colors.py#L2206-L2246

Will release a new version soon.

**Comment by zxdawn** [comment]
@lukelbd Thanks. Works well:
![midpoint](https://user-images.githubusercontent.com/30388627/81494361-e2928380-92da-11ea-902a-70cf0058fc28.png)
